### DISCUSSION TITLE
MS Edge support?

### DISCUSSION THREAD

**Discussion by varogonz95** [lucono/karma-test-explorer#70] [Issues / Help] [answered]
I had some trouble using this extension with MS Edge installed only. It only worked after I installed Chrome.
Is Edge supported?

**Comment by lucono** [comment]
The extension supports any browser that can be configured to work with your Karma project. Chrome is simply the default. Basically, as long as you're able to run your Karma tests with a specific browser directly through Karma (e.g. from the command line) without the extension, then it should also work with the extension.

The extension documentation [here](https://github.com/lucono/karma-test-explorer/blob/master/docs/documentation.md#use-the-browser-option) describes how to do this. Basically:

- Setup your Karma project to work with Edge browser (e.g. using a Karma Edge launcher plugin, or a custom launcher configuration in your Karma config file)
- Verify that Karma is able to run your tests with the browser by running it from the command line
- Use the extensions `karmaTestExplorer.browser` option to let it know to use that browser when running your tests in VS Code

**Reply by varogonz95** [reply]
Thing is, I configured Edge as my browser (via `karma.config.ts`) but still, the extension seems to not be picking it up. I can use the cli though

**Reply by lucono** [reply]
Can you share your karma config showing how Edge was configured?

**Reply by varogonz95** [reply]
Sorry to reply until now.
This is my `karma.config` file
```
// Karma configuration file, see link for more information
// https://karma-runner.github.io/1.0/config/configuration-file.html

module.exports = function (config) {
	config.set({
		basePath: "",
		frameworks: ["jasmine", "@angular-devkit/build-angular"],
		plugins: [
			require("karma-jasmine"),
			require("karma-chrome-launcher"),
			require("@chiragrupani/karma-chromium-edge-launcher"),
			require("karma-jasmine-html-reporter"),
			require("karma-coverage"),
			require("@angular-devkit/build-angular/plugins/karma"),
		],
		client: {
			jasmine: {
				// you can add configuration options for Jasmine here
				// the possible options are listed at https://jasmine.github.io/api/edge/Configuration.html
				// for example, you can disable the random execution with `random: false`
				// or set a specific seed with `seed: 4321`
			},
			clearContext: false, // leave Jasmine Spec Runner output visible in browser
		},
		jasmineHtmlReporter: {
			suppressAll: true, // removes the duplicated traces
		},
		coverageReporter: {
			dir: require("path").join(__dirname, "./coverage/app"),
			subdir: ".",
			reporters: [{ type: "html" }, { type: "text-summary" }],
		},
		reporters: ["progress", "kjhtml"],
		port: 9876,
		colors: true,
		logLevel: config.LOG_INFO,
		autoWatch: true,
		browsers: ["Edge"],
		singleRun: true,
		restartOnFileChange: true,
	});
};
```

**Reply by lucono** [reply, marked as answer]
A workaround you can try is to use the `karmaTestExplorer.env` extension setting to set the `CHROME_BIN` environment variable to the path of the MS Edge executable:

```json
"karmaTestExplorer.env": {
    "CHROME_BIN": "/path/to/microsoft-edge-stable"
}
```
